### PR TITLE
Rideable Carps - Several types of Lassos

### PR DESCRIPTION
**Pull request by DrDuckedGoose**
## About The Pull Request

This is not an exact port! This is an interpretation of someone's suggestion, no external code was referenced, or needs attributing.

Introduces several types of lassos.
- Basic
- Primal
- Antag
- Debug
These lassos can be used to tame animals. Specifically, the basic lasso can tame space-carp & cows, this type can be crafted. The primal lasso can also be crafted, exclusively by ash-walkers, can be used to tame certain lavaland fauna. The antag lasso can tame anything outside of a specified blacklist, bubblegum & other bosses, can be bought from any uplink for 3TC. The debug lasso tames anything.
Tamed animals can be rode & given commands through speech and radial menus(alt-click).
Radial menu icons are procedurally generated, based on the animal's icon.

## Why It's Good For The Game

Unique item, free content.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>
Coder sprites - Subject to change

image
image
image
image
image
image
image
image
image
image
image
image

https://user-images.githubusercontent.com/40559528/177044050-c7368168-31cf-4863-aaca-fd26e0650a47.mp4

</details>

## Changelog
:cl:
add: Adds lasso items
add: Adds tamed riding component subtype, toggles AI & sometimes spacewalking
add: Adds antag lasso to syndicate uplink
add: Add new AI for tamed animals
add: Add tamed component
/:cl:

image

### CODE REVIEW

File: code/game/objects/items/carp_lasso.dm [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/obj/item/carp_lasso
+	name = "carp lasso"
+	desc = "Comes standard with every space-cowboy."
+	icon = 'icons/obj/carp_lasso.dmi'
+	///Ref to timer
+	var/timer
+	///Ref to lasso'd carp
+	var/mob/living/simple_animal/hostile/carp/carp_target
```
Comment (Penwin0) [inline code/game/objects/items/carp_lasso.dm]: Make sure this doesn't cause harddels

Comment (DrDuckedGoose) [reply code/game/objects/items/carp_lasso.dm]: No worries, smokey
image

**Comment by Rukofamicom** [comment]
Are there more implications to this than I'm realizing? It seems like it's just a worse jetpack + the ability to tame an aggressive, but not very threatening mob. Is there anything to the taming process or is it just "use on carp and it's friendly two seconds later"?

Assuming I'm not overlooking something important, what's the possibility of this being added as a general item instead of one exclusive to uplink antags? This seems like a hella fun addition in general and people might actually get excited for carp events.

**Comment by DrDuckedGoose** [comment]
> Are there more implications to this than I'm realizing? It seems like it's just a worse jetpack + the ability to tame an aggressive, but not very threatening mob. Is there anything to the taming process or is it just "use on carp and it's friendly two seconds later"?
> Assuming I'm not overlooking something important, what's the possibility of this being added as a general item instead of one exclusive to uplink antags? This seems like a hella fun addition in general and people might actually get excited for carp events.

That essentially seems to be the case. However, the carps are friendly exclusively to you, making this a general item will generally result in people bringing swathes of aggressive carp to the station.
I didn't want the carp to be friendly to the crew either, my main intentions were allowing people to build carp armies and harass the crew with them.

I can go the other way, and implement it as a general item but, at the same time I would implement a faction inheritance, meaning it would act only as a ghetto jetpack, losing all assault capability.

**Comment by Rukofamicom** [comment]
> I can go the other way, and implement it as a general item but, at the same time I would implement faction inheritance, meaning it would act only as a ghetto jetpack, losing all assault capability.

Any way to make it neutral aggression similar to monkeys or Xenobio slimes? Loyal to owner, but if owner or any other carps also loyal to owner take damage, turn on the person that harmed them?

**Comment by Fronsis** [comment]
Oh, i was under the impression based on [Livrah latest video](https://www.youtube.com/watch?v=O5x2oT96UU0) that you could ride them with the same code as the Attack! Stop! Follow! that they have,  but he didn't clarify on the description i think we fully ported that for Dogs but i guess it also works with.. carps! in any case it's nice knowing that this is an item

as what an old lady friend of mine would say: PRAISE!!!

**Review by Dejaku51 — COMMENTED** [review]
could you change it so it use the AI controller like the dog AI?

**Comment by DrDuckedGoose** [comment]
> could you change it so it use the AI controller like the dog AI?

I recently worked with this AI earlier, for my xenobiology rework proposal. 
https://github.com/BeeStation/BeeStation-Hornet/discussions/7195

It's generally more painful to do but, it also falls into the issue I have regarding speed, regular AI processing just doesn't support the speeds I need to have carps maintain an eloquent speed with the player.
However, if a maintainer yells at me I might.

**Comment by RDS88-beeman** [comment]
Maybe two items, a lasso that gives the carp all your factions for the crew, and a lasso that gives you the carp faction for traitors.

**Comment by PowerfulBacon** [comment]
OK this is cool

**Review by PowerfulBacon — APPROVED** [review]
cool

**Pull request merged**